reactive/etcd.py

```diff
@@ -414,6 +415,35 @@ def initialize_new_leader():
     set_state('etcd.leader.configured')
 
 
+@when('snap.installed.etcd')
+@when('leadership.is_leader')
+def process_snapd_timer():
+    ''' Set the snapd refresh timer on the leader so all cluster members
+    (present and future) will refresh near the same time. '''
+    # Get the current snapd refresh timer (layer-snap will set this).
+    try:
+        timer = snap.get(snapname='core', key='refresh.timer').decode('utf-8')
+    except CalledProcessError:
+        log('snapd_refresh timer not yet set, will retry')
+        return
+
+    # The first time through, data_changed will be true. Subsequent calls
+    # should only update leader data if something changed.
+    if data_changed('etcd_snapd_refresh', timer):
+        log('setting snapd_refresh timer to: {}'.format(timer))
+        leader_set({'snapd_refresh': timer})
+
+
+@when('snap.installed.etcd')
+@when('leadership.changed.snapd_refresh')
+@when_not('leadership.is_leader')
+def set_snapd_timer():
+    ''' Set the snapd refresh.timer on non-leader cluster members. '''
```

Comment:
Do we need to handle where I'm not the leader, but the leader hasn't set snapd_refresh yet?

Reply:
I think we're covered with `@when('leadership.changed.snapd_refresh')`.  If the leader hasn't set it, the leadership.changed flag won't be set.

If you're talking about the case where a follower joins and doesn't know about the leader yet, then yes, layer-snap will set the refresh to a month from yesterday.  But that will quickly be re-set (if it even has a chance to *be* set) as soon as the follower handles the leader hooks.

So, no, i don't think we need a handler for `@when_not(leader)` and `@when_not(leadership.data)`.